docs/commands.md

```diff
@@ -0,0 +1,270 @@
+# Commands
+
+Commands provide a way to interact with and modify a Lean environment outside of the context of a proof.
+Familiar commands from core Lean include `#check`, `#eval`, and `run_cmd`.
+
+Mathlib provides a number of commands that offer customized behavior. These commands fall into two
+categories:
+
+* *Transient* commands are used to query the environment for information, but do not modify it,
+  and have no effect on the following proofs. These are useful as a user to get information from Lean.
+  They should not appear in "finished" files.
+  Transient commands typically begin with the symbol `#`.
```

Comment:
Do we want to include a comment about the transient `set_option pp.*` which do not begin with `#` and shouldn't be in final files?

Reply:
I'm not sure. `set_option` is the command, which is definitely not transient. I think mentioning that might just blur the distinction.